### PR TITLE
Add Usage section to README

### PR DESCRIPTION
**Pull request by Theelx**
This addresses part of #484 by making a short example script which shows how jsonpickle can be used. Todo would be to include an even shorter code snippet in the README itself, as with this change the README just links to the newly-made examples directory in the GitHub repo. This PR also adds some basic utilities for future example scripts, which could showcase numpy extension support, getstate usage, and more advanced topics.

### CODE REVIEW

File: examples/save_class_to_file.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+assert (
+    encoded_instance
+    == '{"py/object": "__main__.Example", "data": {"BAR": 1, "foo": 0}}'
+)
+print(
+    f"jsonpickle successfully encoded the instance of the Example class! It looks like: {encoded_instance}"
+)
+
+
+with open("example.txt", "w+") as f:
```
Comment (davvid) [inline examples/save_class_to_file.py:35]: Should we use `example.json` here and below instead of `.txt`?

File: examples/save_class_to_file.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    f.write(encoded_instance)
+    print(
+        f"jsonpickle successfully wrote the instance of the Example class to example.txt!"
```
Comment (davvid) [inline examples/save_class_to_file.py:38]: We might be able to use a regular string rather than an f-string here.

**Review by davvid — APPROVED** [review]
Good stuff!

File: examples/utilities.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import os
+import sys
+
+
+def ensure_no_files_overwritten(expected_contents):
+    # ensure we don't overwrite the user's files, if they run it on their machine
+    if os.path.exists(os.path.join(os.getcwd(), 'example.txt')):
```
Comment (davvid) [inline examples/utilities.py:7]: We call `open(...)` with just the basename below so we should be able to do the same here and drop the os.path.join parts.

File: examples/utilities.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        with open("example.txt", "r") as f:
+            contents = f.read()
+        if contents != expected_contents:
+            print(
+                "I don't want to overwrite the example.txt file in my current directory! Please delete example.txt and try again!"
+            )
+            # exit with a non-zero error code
+            sys.exit(1)
```
Comment (davvid) [inline examples/utilities.py:15]: A neat trick with `sys.exit` is that you can pass it an error message string and it'll print it out and exit with a non-zero exit code. That lets you combine the `print` and `sys.exit` steps into a single call. It returns exit code 1 as well.

File: examples/save_class_to_file.py (hunk 1 above)
Comment (Theelx) [reply examples/save_class_to_file.py:35]: Yes, good point.

File: examples/save_class_to_file.py (hunk 2 above)
Comment (Theelx) [reply examples/save_class_to_file.py:38]: Opps yeah, I think I previously had something printed out but forgot to remove the f

File: examples/utilities.py (hunk 4 above)
Comment (Theelx) [reply examples/utilities.py:15]: Oh, I didn't know that, thanks!

File: examples/utilities.py (hunk 3 above)
Comment (Theelx) [reply examples/utilities.py]: Yeah fair, I just didn't want to worry about if Windows would throw a fit at doing that. I'll fix that now.

**Pull request merged**